File: src/main/java/net/botlify/anticaptchacom/exceptions/AntiCaptchaComException.java
package net.botlify.anticaptchacom.exceptions;

import lombok.Getter;
import org.jetbrains.annotations.NotNull;

/**
 * An exception thrown by the AntiCaptchaCom API.
 */
public final class AntiCaptchaComException extends Exception {

    /**
     * The code of the response.
     */
    @Getter
    private final int code;

    /**
     * The body of the response.
     */
    @Getter @NotNull
    private final String body;

    public AntiCaptchaComException(final int code,
                                   @NotNull final String body) {
        super("AntiCaptchaCom API error: " + code + " - " + body);
        this.code = code;
        this.body = body;
    }

}

File: src/main/kotlin/net/botlify/anticaptchacom/exceptions/AntiCaptchaComException.kt
package net.botlify.anticaptchacom.exceptions

import com.fasterxml.jackson.annotation.JsonProperty
import net.botlify.anticaptchacom.enums.AntiCaptchaComError

/**
 * An exception thrown by the AntiCaptchaCom API.
 */
data class AntiCaptchaComException internal constructor(
    @JsonProperty("errorId")
    val errorId: Int,
    @JsonProperty("errorCode")
    val errorCode: AntiCaptchaComError,
    @JsonProperty("errorDescription")
    val errorDescription: String
) : Exception(errorDescription)
